cmd/completion.go

```diff
@@ -0,0 +1,253 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/qri-io/ioes"
+	"github.com/spf13/cobra"
+)
+
+// NewAutocompleteCommand creates a new `qri complete` cobra command that prints autocomplete scripts
+func NewAutocompleteCommand(_ Factory, ioStreams ioes.IOStreams) *cobra.Command {
+	o := &AutocompleteOptions{IOStreams: ioStreams}
+	cmd := &cobra.Command{
+		Use:   "completion [bash|zsh]",
+		Short: "Generates shell completion scripts",
+		Long: `To load completion run
+
+source <(qri completion [bash|zsh])
+
+To configure your bash/zsh shell to load completions for each session add to your bashrc/zshrc
+
+# ~/.bashrc or ~/.zshrc
+source <(qri completion [bash|zsh])
+
+Alternatively you can pipe the output to a local script and
+reference that as the source for faster loading
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return o.Run(cmd, args)
+		},
+		ValidArgs: []string{"bash", "zsh"},
+	}
+
+	return cmd
+}
+
+// AutocompleteOptions encapsulates completion options
+type AutocompleteOptions struct {
+	ioes.IOStreams
+}
+
+// Run executes the completion command
+func (o *AutocompleteOptions) Run(cmd *cobra.Command, args []string) (err error) {
+	if len(args) == 0 {
+		return fmt.Errorf("shell not specified")
+	}
+	if len(args) > 1 {
+		return fmt.Errorf("too many arguments, expected only the shell type")
+	}
+	if args[0] == "bash" {
+		cmd.Parent().GenBashCompletion(o.Out)
+	}
+	if args[0] == "zsh" {
+		zshHead := `# reference kubectl completion zsh
```

Comment:
Could we move all of the shell code strings into top-level consts, to keep the golang code straight-forward to read?